plugins/woocommerce-admin/client/products/product-form-actions.tsx

```diff
@@ -73,7 +74,13 @@ export const ProductFormActions: React.FC = () => {
 			...getProductDataForTracks(),
 		} );
 		if ( ! values.id ) {
-			createProductWithStatus( values, 'publish' );
+			const product = await createProductWithStatus( values, 'publish' );
+			if ( product?.id ) {
+				resetForm();
+				navigateTo( {
+					url: 'admin.php?page=wc-admin&path=/product/' + product.id,
+				} );
+			}
```

Comment:
This logic should also be as part of the `onSaveDraft` function right after line 58.

Reply:
oh!!! you are right. It's done now. Thanks @louwie17